src/components/Widget/components/Conversation/components/Messages/index.tsx

```diff
@@ -12,13 +14,22 @@ type Props = {
 }
 
 function Messages({ profileAvatar }: Props) {
-  const messages = useSelector((state: GlobalState) => state.messages.messages);
-  const typing = useSelector((state: GlobalState) => state.behavior.messageLoader);
   const dispatch = useDispatch();
+  const { messages, typing, showChat, badgeCount } = useSelector((state: GlobalState) => ({
+    messages: state.messages.messages,
```

Comment:
I think it's not that necessary. The store initial state will ensure that it will always be at least an empty array. The validation on the JSX it's just for precaution 😄.